mlrun/api/initial_data.py

```diff
@@ -508,33 +541,199 @@ def _rename_marketplace_kind_to_hub(
+        if labels:
+            artifacts_labels_to_migrate.append((new_artifact, labels))
+
+    # commit the new artifacts to the db
+    db._upsert(db_session, new_artifacts, ignore=True)
+
+    # migrate artifact labels to the new table ("artifact_v2_labels")
+    _migrate_artifact_labels(db, db_session, artifacts_labels_to_migrate)
+
+    # migrate artifact tags to the new table ("artifact_v2_tags")
+    _migrate_artifact_tags(db, db_session, artifacts_tags_to_migrate)
+
+
+def _migrate_artifacts_table_v2(
     db: mlrun.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
 ):
-    logger.info("Enriching projects state")
-    projects = db.list_projects(db_session)
-    for project in projects.projects:
-        changed = False
-        if not project.spec.desired_state:
-            changed = True
-            project.spec.desired_state = mlrun.common.schemas.ProjectState.online
-        if not project.status.state:
-            changed = True
-            project.status.state = project.spec.desired_state
-        if changed:
-            logger.debug(
-                "Found project without state data. Enriching",
-                name=project.metadata.name,
+    """
+    Migrate the old artifacts table to the new artifacts_v2 table, including their respective tags and labels.
+    The migration is done in batches, to not overload the db. A state file is used to keep track of the migration
+    progress, and is updated after each batch, so that if the migration fails, it can be resumed from the last batch.
+    Delete the old artifacts table when done.
+    """
+    logger.info("Migrating artifacts to artifacts_v2 table")
+
+    # count the total number of artifacts to migrate
+    total_artifacts_count = db._query(
+        db_session, mlrun.api.db.sqldb.models.Artifact
+    ).count()
+    batch_size = config.artifacts.artifact_migration_batch_size
+
+    # get the last batch that was migrated from the state file
+    batch_index = _get_last_batch_index()
+
+    while batch_index * batch_size < total_artifacts_count:
+        logger.debug(
+            "Migrating artifacts batch",
+            batch_index=batch_index,
+            batch_size=batch_size,
+            total_artifacts_count=total_artifacts_count,
+        )
+        _migrate_artifacts_batch(db, db_session, batch_index, batch_size)
+        batch_index += 1
+        _update_last_batch_index(batch_index)
+
+    # migrate the remaining artifacts
+    _migrate_artifacts_batch(
+        db, db_session, batch_index, total_artifacts_count - batch_index * batch_size
+    )
+
+    # delete the state file
+    _delete_state_file()
+
+    # drop the old artifacts table, including their labels and tags tables
+    db.drop_table(db_session, mlrun.api.db.sqldb.models.Artifact.Label.__tablename__)
+    db.drop_table(db_session, mlrun.api.db.sqldb.models.Artifact.Tag.__tablename__)
+    db.drop_table(db_session, mlrun.api.db.sqldb.models.Artifact.__tablename__)
```

Comment:
This is going to cause an issue when we deprecate the old `Artifacts` table. 
You'll have DBs in the field without this table, then once we remove the table from the sqlalchemy models the resulting migration will also try to drop the table (as it should if the database is new and running all migrations to just get to the latest version).
Either:
- Empty the tables here but don't drop them.
- Add a TODO here to remember to edit the resulting migration when deprecated to make the drop table in the migration be ok with the table not existing at all.